src/SmartFormat/SmartFormatter.cs

```diff
@@ -182,10 +219,24 @@ public void FormatInto(IOutput output, string format, params object[] args)
         /// <returns>Returns the formatted input with items replaced with their string representation.</returns>
         public string FormatWithCache(ref FormatCache? cache, string format, params object[] args)
         {
-            var output = new StringOutput(format.Length + args.Length * 8);
+            return FormatWithCache(ref cache, args, format);
+        }
+
+        /// <summary>
+        /// Replaces one or more format items in a specified string with the string representation of a specific object,
+        /// using the <see cref="FormatCache"/>.
+        /// </summary>
+        /// <param name="cache">The <see cref="FormatCache" /> to use.</param>
+        /// <param name="args">The objects to format.</param>
+        /// <param name="format">A composite format string.</param>
+        /// <returns>Returns the formatted input with items replaced with their string representation.</returns>
+        public string FormatWithCache(ref FormatCache? cache, IList<object> args, string format)
+        {
+            args = args ?? k_Empty;
```

Comment:
Should read:

```args ??= k_Empty;```

in case args is `nullable` - otherwise this line is redundant

Reply:
Ah yes our version of dotnet does not support that.